src/lsqecc/pauli_rotations/rotation.py

```diff
@@ -192,6 +192,43 @@ def from_list(pauli_ops: List[PauliOperator], rotation: Fraction) -> "PauliRotat
             r.change_single_op(i, op)
         return r
 
+    def remove_y_operators(self) -> Tuple[List["PauliRotation"], List["PauliRotation"]]:
```

Comment:
This is already a huge improvement in terms of engineering. I can only suggest going even a step further. Just as a good software heuristic, I suggest considering to apply the principle of least privilege and making this method a non modifier. Something like 
```py
def get_y_operator_free_equivalent(self) -> List["PauliRotation"]:
```
Of course usage in the circuit would have to change as well.

Reply:
Just updated the code based on the suggestion. Can you have a look?

Reply:
Looks good, this is exactly what I meant. I'm assuming you will integrate them in Circuit as well?

Reply:
Did it in the last commit, also made the circuit method to be non-modifier. Might need a name update and update to the codes that are calling it.